apm-lambda-extension/extension/route_handlers.go

```diff
@@ -31,30 +31,31 @@ type AgentData struct {
 	ContentEncoding string
 }
 
-var AgentDoneSignal chan struct{}
-var mainExtensionContext context.Context
+var currApmServerTransport *ApmServerTransport
+var currContext context.Context
 
 // URL: http://server/
-func handleInfoRequest(ctx context.Context, apmServerUrl string, config *extensionConfig) func(w http.ResponseWriter, r *http.Request) {
+func handleInfoRequest(ctx context.Context, apmServerTransport *ApmServerTransport) func(w http.ResponseWriter, r *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 
-		Log.Debug("Handling APM Server Info Request")
-		mainExtensionContext = ctx
+		Log.Debug("Handling APM server Info Request")
 
 		// Init reverse proxy
-		parsedApmServerUrl, err := url.Parse(apmServerUrl)
+		parsedApmServerUrl, err := url.Parse(apmServerTransport.config.apmServerUrl)
 		if err != nil {
 			Log.Errorf("could not parse APM server URL: %v", err)
 			return
 		}
 
 		reverseProxy := httputil.NewSingleHostReverseProxy(parsedApmServerUrl)
 
-		reverseProxyTimeout := time.Duration(config.DataForwarderTimeoutSeconds) * time.Second
+		reverseProxyTimeout := time.Duration(apmServerTransport.config.DataForwarderTimeoutSeconds) * time.Second
 		customTransport := http.DefaultTransport.(*http.Transport).Clone()
 		customTransport.ResponseHeaderTimeout = reverseProxyTimeout
 		reverseProxy.Transport = customTransport
 
+		currApmServerTransport = apmServerTransport
+		currContext = ctx
 		reverseProxy.ErrorHandler = reverseProxyErrorHandler
```

Comment:
```suggestion
		reverseProxy.ErrorHandler = func(w http.ResponseWriter, r *http.Request, err error) {
			SetApmServerTransportState(ctx, apmServerTransport, Failing)
			Log.Errorf("Error querying version from the APM server: %v", err)
		}
```

Then we don't need to store globals.

Reply:
Implemented in 457e247